packages/salesforcedx-vscode-core/src/commands/forceProjectCreate.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import * as glob from 'glob';
+import * as path from 'path';
+import { Observable } from 'rxjs/Observable';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
+import { nls } from '../messages';
+import { notificationService } from '../notifications';
+import { CancellableStatusBar, taskViewService } from '../statuses';
+import {
+  CancelResponse,
+  CompositeParametersGatherer,
+  ContinueResponse,
+  EmptyPreChecker,
+  ParametersGatherer,
+  PostconditionChecker,
+  SfdxCommandlet,
+  SfdxCommandletExecutor
+} from './commands';
+
+export class ForceProjectCreateExecutor extends SfdxCommandletExecutor<{}> {
+  public build(data: ProjectNameAndPath): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_project_create_text'))
+      .withArg('force:project:create')
+      .withFlag('--projectname', data.projectName)
+      .withFlag('--outputdir', data.projectUri)
+      .build();
+  }
+
+  public execute(response: ContinueResponse<ProjectNameAndPath>): void {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: response.data.projectUri
+    }).execute(cancellationToken);
+
+    execution.processExitSubject.subscribe(async data => {
+      if (data != undefined && data.toString() === '0') {
+        await vscode.commands.executeCommand(
+          'vscode.openFolder',
+          vscode.Uri.parse(
+            path.join(response.data.projectUri, response.data.projectName)
+          )
+        );
+      }
+    });
+
+    notificationService.reportExecutionError(
+      execution.command.toString(),
+      (execution.stderrSubject as any) as Observable<Error | undefined>
+    );
+    channelService.streamCommandOutput(execution);
+    CancellableStatusBar.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+}
+
+export type ProjectNameAndPath = ProjectName & ProjectURI;
+
+export interface ProjectURI {
+  projectUri: string;
+}
+
+export interface ProjectName {
+  projectName: string;
+}
+
+export class SelectProjectName implements ParametersGatherer<ProjectName> {
```

Comment:
Since these ParametersGatherers and Condition checker are currently only used in this class, I am leaving them here. If you would prefer to move them to `commands.ts`, please comment and let me know.